### ISSUE TITLE
errors in shell-execute

### ISSUE THREAD

**Issue by knudje** [pimatic/pimatic#353]
Hi,
In the logs, I found a lot of errors with shell execute: see below.
Any ideas?

Possibly unhandled ReferenceError: name is not defined
  at /home/pi/pimatic-app/node_modules/pimatic-shell-execute/shell-execute.coffee:96:32
  at tryCatch1 (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/main/util.js:43:21)
  at Promise$_callHandler [as _callHandler] (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/main/promise.js:627:13)
  at Promise$_settlePromiseFromHandler [as _settlePromiseFromHandler] (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/main/promise.js:641:18)
  at Promise$_settlePromiseAt [as _settlePromiseAt] (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/main/promise.js:800:14)
  at Promise$_settlePromises [as _settlePromises] (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/main/promise.js:934:14)
  at Async$_consumeFunctionBuffer [as _consumeFunctionBuffer] (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/main/async.js:75:12)
  at Async$consumeFunctionBuffer (/home/pi/pimatic-app/node_modules/pimatic/node_modules/bluebird/js/main/async.js:38:14)
  at process._tickCallback (node.js:415:13)

**Comment by sweetpi** [comment]
Closed because of inactivity.

**Comment by Heizelmann** [comment]
Actually I have the same error with version 0.8.5. 
```
20:57:13.227 [pimatic] debug: ReferenceError: name is not defined
20:57:13.227 [pimatic] debug:>    at /home/pi/pimatic-app/node_modules/pimatic-shell-execute/shell-execute.coffee:167:67
```

The error arrises when a shell command returns with an unexpexted result. Probably the coffee script code is wrong or is compiled wrong to javascript. The codeline in shell-execute-coffee is
```
         throw new Error("Error getting attribute vale for #{name}: #{stderr}")
```
and is compiled to javascript
```
            throw new Error("Error getting attribute value for " + name + ": " + stderr);
``` 
I am not an expert in coffe script, so I changed temporary the javascript to
```
            throw new Error("Error getting attribute value for " + _this.name + ": " + stderr);
```
and the error is solved. Now the log result is somwhat like this
```
Possibly unhandled Error: Error getting attribute value for MyTestShellSensor: SRV ERR: command unknown

    at /shell-execute.coffee:110:21
```
which is the expected output for my test ShellSensor

**Comment by sweetpi** [comment]
@Heizelmann Thanks for pointing this out. Should be fixed in pimatic-shell-execute: v0.8.6

**Comment by Heizelmann** [comment]
@sweetpi Thanks for the quick reaction!
